src/components/create-study-form.js

```diff
@@ -285,31 +319,41 @@ export const CreateStudyForm = ({ open, onClose }) => {
             selectedFile,
             isPrivateStudy,
             activeDirectory
-        ).then((res) => {
-            if (res.ok) {
-                onClose();
-                resetDialog();
-            } else {
-                console.debug('Error when creating the study');
-                if (res.status === 409) {
-                    setCreateStudyErr(
-                        intl.formatMessage({ id: 'studyNameAlreadyUsed' })
-                    );
-                } else {
-                    res.json()
-                        .then((data) => {
-                            setCreateStudyErr(
-                                data.error + ' - ' + data.message
-                            );
-                        })
-                        .catch((error) => {
-                            setCreateStudyErr(
-                                error.name + ' - ' + error.message
-                            );
-                        });
+        )
+            .then((res) => {
+                dispatch(removeUploadingStudy(uploadingStudy));
+                if (!res.ok) {
+                    if (res.status === 409) {
+                        studyCreationError(
+                            studyName,
+                            intl.formatMessage({
+                                id: 'studyNameAlreadyUsed',
+                            })
+                        );
+                    } else {
+                        res.json()
+                            .then((data) => {
+                                studyCreationError(
+                                    studyName,
+                                    data.error + ' - ' + data.message
+                                );
+                            })
+                            .catch((error) => {
+                                studyCreationError(
+                                    studyName,
+                                    error.name + ' - ' + error.message
+                                );
+                            });
+                    }
                 }
-            }
-        });
+            })
+            .catch((e) => {
+                studyCreationError(studyName, e.name + ' - ' + e.message);
+                dispatch(removeUploadingStudy(uploadingStudy));
+            });
+        dispatch(addUploadingStudy(uploadingStudy));
```

Comment:
Could you move this before `createStudy` call ? I know it doesn't change anything with the Promise here which code is executed asynchronously, but it's more logical to `addUploadingStudy` first then `RemoveUploadingStudy`

Reply:
the call is not done, it is not uploading